dynaconf/utils/parse_conf.py

```diff
@@ -276,6 +276,20 @@ def get_converter(converter_key, value, box_settings):
     return converted_value
 
 
+def add_converter(converter_key, casting_callable):
+    global converters
+
+    def converter_callable(value, box_settings=None):
+        if isinstance(value, Lazy):
+            # value got here as a Lazy object, needs evaluation
+            return Lazy(value(box_settings), casting=casting_callable)
+        else:
+            # value is good to go
+            return Lazy(value, casting=casting_callable)
```

Comment:
Would be good to set a `formatter` here, the default is `str.format` which works fine but we can avoid that with

```py
def noop_formatter_factory(converter_key):
    """Builds a new noop formatter and sets its token"""
    return BaseFormatter(lambda x, **_: x, converter_key)
```

and

```suggestion
            return Lazy(
                value, 
                casting=casting_callable, 
                formatter=noop_formatter_factory(f"@{converter_key}")
            )
```